twine/package.py

```diff
@@ -197,6 +202,21 @@ def metadata_dictionary(self) -> Dict[str, MetadataValue]:
 
         return data
 
+    def add_attestations(self, attestations: List[str]) -> None:
+        loaded_attestations = []
+        for attestation in attestations:
+            with open(attestation, "rb") as att:
+                raw_attestation = att.read()
+
+            try:
+                loaded_attestations.append(json.loads(raw_attestation))
```

Comment:
JSON load can take a file so we're not reading everything into memory. I don't think it would complicate this that much and then the JSON can be read and parsed a bit more efficiently

Reply:
Fixed, thanks!